tests/integration/any_charm.py

```diff
@@ -41,6 +44,49 @@ def update_ingress(self, ingress_config):
         """
         self.ingress.update_config(ingress_config)
 
+    def _has_required_fields(self, rel: Dict) -> bool:
+        """Check for required fields in relation
+
+        Args:
+            rel: relation to check
+
+        Returns:
+            Returns true if all fields exist
+        """
+        return "service-hostname" in rel and "service-name" in rel and "service-port" in rel
```

Comment:
How about `return all(key in rel for key in ("service-hostname", "service-name", "service-port"))`?